### PR TITLE
Enable analytics

### PR DESCRIPTION
**Pull request by florian-richter**
## Proposed Changes

Generate the necessary file (if given consent) to send analytics.

## Type of Changes

- [ ] Bug Fix
- [x] New Feature
- [ ] Documentation Update

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

## Breaking Changes

In general avoid breaking changes, but if you think it is necessary give your reasoning in the _Further Comments_ section.

- [x] I have not introduced breaking changes

## Further Comments

If the changes are complex, please discuss what alternatives you have evaluated and why you picked this particular solution. Please make sure to highlight any trade-offs that we should know about.

### CODE REVIEW

File: src/commands/init.ts [hunk 1]
```diff
@@ -30,6 +40,10 @@ export default class Init extends Command {
       hidden: true,
       description: 'If the folder is empty, use nest-cli to create a project scaffold.'
     }),
+    analytics: flags.boolean({
+      hidden: true,
+      description: 'Set analytics as enabled or disabled.'
```
Comment (marikaner) [inline src/commands/init.ts]: "Enable collection of anonymous usage data."

File: src/commands/init.ts [hunk 2]
```diff
@@ -106,7 +120,7 @@ export default class Init extends Command {
         },
         {
           title: 'Installing dependencies',
-          task: () => installDependencies(projectDir, verbose).catch(e => this.error(`Error during npm install: ${e.message}`, { exit: 2 }))
+          task: () => installDependencies(projectDir, verbose).catch(e => this.error(`Error during npm install: ${e.message}`, { exit: 13 }))
```
Comment (marikaner) [inline src/commands/init.ts]: I had to google to know what 13 means. Can we put an enum somwhere? I didn't find a lib for that, but still...
https://github.com/nodejs/node-v0.x-archive/blob/master/doc/api/process.markdown#exit-codes

File: src/utils/usage-analytics.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import cli from 'cli-ux';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export async function usageAnalytics(projectDir: string, agreeToAnalytics: boolean) {
+  if (agreeToAnalytics === false) {
+    return;
+  }
+  if (agreeToAnalytics || (await cli.confirm('Do you want to provide anonymous usage analytics to help us improve the SDK? (y|n)'))) {
```
Comment (marikaner) [inline src/utils/usage-analytics.ts]:
```suggestion
  if (agreeToAnalytics !== false || (await cli.confirm('Do you want to provide anonymous usage analytics to help us improve the SDK? (y|n)'))) {
```

Then you could remove the first if.

Comment (florian-richter) [reply src/utils/usage-analytics.ts]: This is intentional. The suggestion would change the behavior of the if statement

File: src/commands/init.ts (hunk 2 above)
Comment (florian-richter) [reply src/commands/init.ts]: Exit codes are just counting up from 10 (more or less) so we can identify where an error happened

**Pull request merged**